python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -1197,6 +1221,115 @@ async def set_hexapod(self, azimuth, elevation, axis=f'xyzuvw'):
+            # 10x smaller than that is surely noise.
+
+            if abs(self.focusOffsetYetToBeApplied) > abs(self.correction_tolerance['z'] / 10):
+                self.log.info(
+                    f'Applying focus offset of {self.focusOffsetYetToBeApplied} '
+                    f'in correction loop due to filter '
+                    f'and/or disperser changes.')
+                # add the offset, then reset the value
+                # using subtraction here to avoid a possible race condition
+                # note that self.focusOffsetYetToBeApplied is a value not an object
+                # so no deepcopy is required
+                _offsetValue = copy.deepcopy(self.focusOffsetYetToBeApplied)
+                self.model.add_offset("z", _offsetValue)
+                self.focusOffsetYetToBeApplied -= _offsetValue
+                # publish events with new offsets
+                self.evt_correctionOffsets.set_put(**self.model.offset,
+                                                   force_output=True)
+                # Do accounting to republish total offset and others that were set
+                # in the callbacks
+                self.focusOffsetPerCategory['total'] = self.model.offset['z']
+                self.evt_focusOffsetSummary.set_put(**self.focusOffsetPerCategory,
+                                                    force_output=True)
+                await asyncio.sleep(0)
+
+            # Now do pointing
+            if abs(np.max(self.pointingOffsetsYetToBeApplied)) > abs(_pointingTolerance):
+                self.log.info(
+                    f'Applying pointing offset of [X,Y]={self.pointingOffsetsYetToBeApplied} '
+                    f'in correction loop due to filter '
+                    f'and/or disperser changes.')
+                _pointingOffsets = copy.deepcopy(self.pointingOffsetsYetToBeApplied)
+            elif(_offsetValue):
+                # No offsets above thresholds, so just return
+                self.log.debug('Focus and pointing offsets below tolerances. Passing without correcting')
+                pass
+
+            # Corrections required, so flip the bit on the detailed state
+            self.detailed_state = self.detailed_state ^ status_bit
+
+            self.log.debug(f'_offsetValue is {_offsetValue}')
+            self.log.debug(f'_pointingOffsets is {_pointingOffsets}')
+
+            # send out even saying correction is started
+            self.evt_atspectrographCorrectionStarted.set_put(focusOffset=_offsetValue,
+                                                             pointingOffsets=_pointingOffsets)
+
+            try:
+                # Don't need a tolerance since these values only get set if they meet a tolerance already
+                if abs(_offsetValue):
+                    self.log.debug('Applying focus correction with hexapod')
+                    await self.set_hexapod(azimuth, elevation)
+                    await asyncio.sleep(0)
```

Comment:
you don't need a `await asyncio.sleep(0)` after a `await`... this should be added if you spend some time without any call to `await`.